babe

```diff
@@ -201,12 +218,68 @@ def destroyCommand(destroyAppOptionsParser):
     appName = destroyAppOptionsParser.app
     destroyApp(appName)
 
+def getMesosSlaveId(mesosMasterEndpoint, hostname):
+    r = requests.get(endpoint_url(mesosMasterEndpoint, '/slaves'), auth=httpBasicCreds, verify=httpsVerify)
+    if 200 == r.status_code:
+        slaves = [slave['id'] for slave in r.json()['slaves'] if slave['hostname'] == hostname and slave['active'] == True]
+        if not slaves:
+            print("No matching active mesos slave with hostname {0}".format(hostname))
+            return
+        elif len(slaves) > 1:
+            print("Multiple active mesos slaves with hostname {0}".format(hostname))
+            return
+        else:
+            return slaves[0]
+
+def getAppInstances(appName):
+    r = getAppInfo(appName)
+    if None == r:
+        print("No app: " + appName)
+        return
+    data=r.json()
+    app=data['app']
+    instances = app['tasks']
+    return instances
+
+def getTaskInstance(task, filter):
+    appInstances = getAppInstances(task['appId'])
+    if filter(appInstances):
+        return {'appName': task['appId'], 'taskId': task['id'], 'instance': task}
+
+def decommission(mesosMasterEndpoint, hostname):
+    slaveId = getMesosSlaveId(mesosMasterEndpoint, hostname)
+    if not slaveId:
+        print("Cannot resolve slaveId for hostname {0}".format(hostname))
+        return
+    prompt_response = raw_input("Do you really want to decommission node with hostname = {0}, slaveId = {1} [y/n]? ".format(hostname, slaveId))
+    if prompt_response == 'y':
+        r = requests.get(url("/v2/tasks"), auth=httpBasicCreds, verify=httpsVerify)
+        if 200 == r.status_code:
+            print("Fetching list of tasks that need to be killed....")
+            tasks = [getTaskInstance(task, lambda l: len(l) > 1) for task in r.json()['tasks'] if task['slaveId'] == slaveId]
+            tasks = [task for task in tasks if task is not None]
```

Comment:
Compile a list of apps that has only one instance and it is running only on this host, just report the list properly and exit from here asking the user to scale them up first.